ray-operator/controllers/ray/common/service_test.go

```diff
@@ -235,38 +235,51 @@ func TestGetPortsFromCluster(t *testing.T) {
 func TestGetServicePortsWithMetricsPort(t *testing.T) {
 	cluster := instanceWithWrongSvc.DeepCopy()
 
-	// Test case 1: No ports are specified by the user.
-	cluster.Spec.HeadGroupSpec.Template.Spec.Containers[0].Ports = []corev1.ContainerPort{}
-	ports := getServicePorts(*cluster)
-	// Verify that getServicePorts sets the default metrics port when the user doesn't specify any ports.
-	if ports[utils.MetricsPortName] != int32(utils.DefaultMetricsPort) {
-		t.Fatalf("Expected `%v` but got `%v`", int32(utils.DefaultMetricsPort), ports[utils.MetricsPortName])
-	}
-
-	// Test case 2: Only a random port is specified by the user.
-	cluster.Spec.HeadGroupSpec.Template.Spec.Containers[0].Ports = []corev1.ContainerPort{
+	testCases := []struct {
+		modifyPorts  func(*rayv1.RayCluster)
```

Comment:
```suggestion
		ports  []corev1.ContainerPort
```

And then in the for loop below, use `cluster.DeepCopy` followed with `cluster.Spec.HeadGroupSpec.Template.Spec.Containers[0].Ports = testCase.ports`